verify.js

```diff
@@ -59,23 +59,9 @@ module.exports = function (jwtString, secretOrPublicKey, options, callback) {
     return done(new JsonWebTokenError('secret or public key must be provided'));
```

Comment:
We should move these ^ asserts inside the `getSecrect` callback too, since you could have `secretOrPublicKey` here but it could be a function that returns value / no-value later on. I mean:
```
   var hasSignature = parts[2].trim() !== '';
 	 
   if (!hasSignature && secretOrPublicKey){
     return done(new JsonWebTokenError('jwt signature is required'));
   }
 	 
   if (hasSignature && !secretOrPublicKey) {
     return done(new JsonWebTokenError('secret or public key must be provided'));
   }
```

Reply:
I do believe, as the other PR-owner, that the jws-package already handles these cases. But will move them anyway ;-)